test/utils/addUser.js

```diff
@@ -12,7 +12,19 @@ module.exports = async (user) => {
   const isValid = user && Object.keys(user).length !== 0 && user.constructor === Object;
   // Use the user data sent as arguments, else use data from fixtures
   user = isValid ? user : userData[0];
+  /*
+    Making a shallow copy of user allows us to safely modify the user object
+    without affecting the original fixture
+  */
+  user = { ...user };
+  const roles = user.roles;
+  delete user.roles;
   const { userId } = await users.addOrUpdate(user);
-
+  if (roles) {
+    const persistedUser = (await users.fetchUser({ userId })).user;
+    const currentRoles = persistedUser.roles;
+    const updatedRoles = currentRoles ? { ...currentRoles, ...roles } : roles;
```

Comment:
Will this work if `currentRoles` is an empty object?

Reply:
Yes. It'll work. Removed the ternary